controllers/component_build_controller.go

```diff
@@ -543,8 +549,37 @@ func ConfigureRepositoryForPaC(component appstudiov1alpha1.Component, config map
 		return prUrl, nil
 
 	case "gitlab":
-		// TODO implement
-		return "", fmt.Errorf("git provider %s is not supported", gitProvider)
+		glclient, err := gitlab.NewGitlabClient(accessToken)
+		if err != nil {
+			return "", err
+		}
+
+		gitlabNamespace := gitSourceUrlParts[3]
+		gitlabProjectName := gitSourceUrlParts[4]
+		projectPath := gitlabNamespace + "/" + gitlabProjectName
+
+		err = gitlab.SetupPaCWebhook(glclient, projectPath, webhookTargetUrl, webhookSecret)
+		if err != nil {
+			return "", err
+		}
+
+		mrData := &gitlab.PaCMergeRequestData{
+			ProjectPath:   projectPath,
+			CommitMessage: commitMessage,
+			Branch:        branch,
+			BaseBranch:    baseBranch,
+			MrTitle:       mrTitle,
+			MrText:        mrText,
+			AuthorName:    authorName,
+			AuthorEmail:   authorEmail,
+			Files: []gitlab.File{
+				{FullPath: ".tekton/" + component.Name + "-" + PipelineRunOnPushFilename, Content: pipelineOnPush},
+				{FullPath: ".tekton/" + component.Name + "-" + PipelineRunOnPRFilename, Content: pipelineOnPR},
+			},
+		}
+		mrUrl, err := gitlab.EnsurePaCMergeRequest(glclient, mrData)
```

Comment:
similar validation of input question here

Reply:
Merge request data struct is filled by the operator based on the component fields (name) and git repository url (user name, repository name). If check for git repository url correctness is added, then we are 100% safe here.

Reply:
Moreover, if the git repository url is invalid, I think application service will fail first (on detecting devfile), so build service will not start it's work.
However, I'd still have the check as the logic could be changed one day.

Reply:
sounds good ... feel free to resolve based on your updates per the comment thread above